### ISSUE TITLE
CI can pass even with type errors in the build

### ISSUE THREAD

**Issue by samestep** [penrose/penrose#866]
Example on commit 1810251d659e2c261c351212fcbc2e6e7dfad36e: https://github.com/penrose/penrose/runs/4873069718?check_suite_focus=true

Selected lines from the log:

```
@penrose/browser-ui: src/App.tsx(223,24): error TS2554: Expected 4 arguments, but got 3.
@penrose/panels: src/App.tsx(162,26): error TS2554: Expected 4 arguments, but got 3.
@penrose/synthesizer-ui: src/components/Gridbox.tsx(145,17): error TS2554: Expected 4 arguments, but got 3.
```

**Comment by wodeni** [comment]
Strange. Have you tried running `yarn build` in those packages individually? Could be an estrella glitch tbh.

**Comment by samestep** [comment]
@wodeni `yarn build` fails locally in each of those three packages on that commit, but `yarn build` from the repo root passes even if I set `CI=1`.

**Comment by wodeni** [comment]
Umm... do you think this could be lerna? I tried inserting a type error and it worked as expected though. Anything special about this?

**Comment by samestep** [comment]
@wodeni This issue still persists. Example:

- https://github.com/penrose/penrose/pull/907/commits/cdef6952271bc6d9a973d34d1b2eb61abf22d032
- https://github.com/penrose/penrose/runs/6132954177?check_suite_focus=true

```
@penrose/browser-ui: $ node build.js
@penrose/browser-ui: Wrote public/app.js (13.1MB, 1.4s)
@penrose/browser-ui: Waiting for TypeScript... (^C to skip)
@penrose/browser-ui: src/inspector/views/GraphUtils.ts(37,21): error TS2339: Property 'id' does not exist on type 'VarAD'.
@penrose/browser-ui:   Property 'id' does not exist on type 'number'.
@penrose/browser-ui: src/inspector/views/GraphUtils.ts(37,47): error TS2339: Property 'op' does not exist on type 'VarAD'.
@penrose/browser-ui:   Property 'op' does not exist on type 'number'.
@penrose/browser-ui: src/inspector/views/GraphUtils.ts(41,39): error TS2339: Property 'childrenAD' does not exist on type 'VarAD'.
@penrose/browser-ui:   Property 'childrenAD' does not exist on type 'number'.
@penrose/browser-ui: src/inspector/views/GraphUtils.ts(50,[25](https://github.com/penrose/penrose/runs/6132954177?check_suite_focus=true#step:4:25)): error TS2339: Property 'childrenAD' does not exist on type 'VarAD'.
@penrose/browser-ui:   Property 'childrenAD' does not exist on type 'number'.
@penrose/browser-ui: src/inspector/views/GraphUtils.ts(57,29): error TS[27](https://github.com/penrose/penrose/runs/6132954177?check_suite_focus=true#step:4:27)69: No overload matches this call.
@penrose/browser-ui:   Overload 1 of 2, '(...items: ConcatArray<NodeDefinition>[]): NodeDefinition[]', gave the following error.
@penrose/browser-ui:     Argument of type 'unknown[]' is not assignable to parameter of type 'ConcatArray<NodeDefinition>'.
@penrose/browser-ui:       The types returned by 'slice(...)' are incompatible between these types.
@penrose/browser-ui:         Type 'unknown[]' is not assignable to type 'NodeDefinition[]'.
@penrose/browser-ui:           Type 'unknown' is not assignable to type 'NodeDefinition'.
@penrose/browser-ui:   Overload 2 of 2, '(...items: (NodeDefinition | ConcatArray<NodeDefinition>)[]): NodeDefinition[]', gave the following error.
@penrose/browser-ui:     Argument of type 'unknown[]' is not assignable to parameter of type 'NodeDefinition | ConcatArray<NodeDefinition>'.
@penrose/browser-ui:       Type 'unknown[]' is not assignable to type 'ConcatArray<NodeDefinition>'.
@penrose/browser-ui: src/inspector/views/GraphUtils.ts(58,25): error TS2769: No overload matches this call.
@penrose/browser-ui:   Overload 1 of 2, '(...items: ConcatArray<EdgeDefinition>[]): EdgeDefinition[]', gave the following error.
@penrose/browser-ui:     Argument of type 'unknown[]' is not assignable to parameter of type 'ConcatArray<EdgeDefinition>'.
@penrose/browser-ui:       The types returned by 'slice(...)' are incompatible between these types.
@penrose/browser-ui:         Type 'unknown[]' is not assignable to type 'EdgeDefinition[]'.
@penrose/browser-ui:           Type 'unknown' is not assignable to type 'EdgeDefinition'.
@penrose/browser-ui:   Overload 2 of 2, '(...items: (EdgeDefinition | ConcatArray<EdgeDefinition>)[]): EdgeDefinition[]', gave the following error.
@penrose/browser-ui:     Argument of type 'unknown[]' is not assignable to parameter of type 'EdgeDefinition | ConcatArray<EdgeDefinition>'.
@penrose/browser-ui:       Type 'unknown[]' is not assignable to type 'ConcatArray<EdgeDefinition>'.
```

**Comment by samestep** [comment]
@wodeni Is this fixed now as of #942?
